src/utils/openai.ts

```diff
@@ -97,17 +98,78 @@ const sanitizeMessage = (message: string) => message.trim().replace(/[\n\r]/g, '
 
 const deduplicateMessages = (array: string[]) => Array.from(new Set(array));
 
-const getPrompt = (locale: string, diff: string) => `Write an insightful but concise Git commit message in a complete sentence in present tense for the following diff without prefacing it with anything, the response must be in the language ${locale}:\n${diff}`;
+const getBasePrompt = (locale: string) => `Write an insightful but concise Git commit message in a complete sentence in present tense for the diff that I provide you without prefacing it with anything, the response must be in the language ${locale}`;
+
+const getCommitMessageFormatPrompt = (type: CommitType) => {
+	if (type === 'conventional') {
+		return '<type>(<optional scope>): <commit message>';
+	}
+
+	return '<commit message>';
+};
+
+const getExtraContextForConventionalCommits = () => {
+	/**
+	 * References:
+	 * Commitlint:
+	 * https://github.com/conventional-changelog/commitlint/blob/18fbed7ea86ac0ec9d5449b4979b762ec4305a92/%40commitlint/config-conventional/index.js#L40-L100
+	 *
+	 * Conventional Changelog:
+	 * https://github.com/conventional-changelog/conventional-changelog/blob/d0e5d5926c8addba74bc962553dd8bcfba90e228/packages/conventional-changelog-conventionalcommits/writer-opts.js#L182-L193
+	 */
+	const conventionalCommitTypes: Record<string, string> = {
+		/*
+			Commented out because they are too common and
+			will cause the model to generate them too often.
+		*/
+		// feat: 'A new feature',
+		// fix: 'A bug fix',
```

Comment:
Isn't it safe to uncomment these now that Conventional Commits aren't mentioned?